packages/host/app/components/operator-mode/stack-item.gts

```diff
@@ -204,64 +203,80 @@ export default class OperatorModeStackItem extends Component<Signature> {
     await navigator.clipboard.writeText(this.card.id);
   });
 
-  fetchRealmInfo = trackedFunction(
+  private fetchRealmInfo = trackedFunction(
     this,
     async () => await this.cardService.getRealmInfo(this.card),
   );
 
-  clearSelections = () => {
+  private clearSelections = () => {
     this.selectedCards.splice(0, this.selectedCards.length);
   };
 
   @cached
-  get iconURL() {
+  private get iconURL() {
     return this.fetchRealmInfo.value?.iconURL;
   }
 
-  get realmName() {
+  private get realmName() {
     return this.fetchRealmInfo.value?.name;
   }
 
-  get cardIdentifier() {
+  private get cardIdentifier() {
     return this.args.item.url?.href;
   }
 
   @action
-  hoverOnRealmIcon() {
+  private hoverOnRealmIcon() {
     this.isHoverOnRealmIcon = !this.isHoverOnRealmIcon;
   }
 
-  get headerIcon() {
+  private get headerIcon() {
     return {
       URL: this.iconURL,
       onMouseEnter: this.hoverOnRealmIcon,
       onMouseLeave: this.hoverOnRealmIcon,
     };
   }
 
-  get headerTitle() {
+  private get headerTitle() {
     return this.isHoverOnRealmIcon && this.realmName
       ? `In ${this.realmName}`
       : cardTypeDisplayName(this.card);
   }
 
-  get canWrite() {
+  private get canWrite() {
     return this.realmResource?.canWrite;
   }
 
+  private get moreOptionsMenuItems() {
+    let menuItems: MenuItem[] = [
+      new MenuItem('Copy Card URL', 'action', {
+        action: () => this.copyToClipboard.perform(),
+        icon: IconLink,
+        disabled: !this.card.id,
+      }),
+    ];
+    if (this.canWrite) {
+      menuItems.push(
+        new MenuItem('Delete', 'action', {
+          action: () => this.args.publicAPI.delete(this.card),
+          icon: IconTrash,
+          dangerous: true,
+          disabled: !this.card.id,
+        }),
+      );
+    }
+    return menuItems;
+  }
+
   @cached
   get card(): CardDef {
     return this.args.item.card;
   }
 
   private loadCard = restartableTask(async () => {
     await this.args.item.ready();
-    // in the case where we get no realm URL from the card, we are dealing with
-    // a new card instance that does not have a realm URL yet. For now let's
-    // assume that the new card instance will reside in the realm that is hosting the app...
-    let realmURL =
-      (await this.cardService.getRealmURL(this.card)) ?? new URL(ownRealmURL);
-    this.realmResource = getRealm(this, () => realmURL);
+    this.realmResource = getRealm(this, { card: () => this.card });
```

Comment:
i made an update to the RealmResource to also get a realm resource based on the realm a card belongs to, so that the old logic that was here wouldn't get duplicated a lot